osquery/tables/system/darwin/user_groups.mm

```diff
@@ -55,6 +56,60 @@ void genODEntries(ODRecordType type, std::set<std::string> &names) {
   }
 }
 
+void genPolicyColumns(int uid, boost::property_tree::ptree& tree) {
+  @autoreleasepool {
+    ODSession* s = [ODSession defaultSession];
+    NSError* err = nullptr;
+    ODNode* root = [ODNode nodeWithSession:s name:@"/Local/Default" error:&err];
+    if (err != nullptr) {
+      TLOG << "Error with OpenDirectory node: "
+           << std::string([[err localizedDescription] UTF8String]);
+    }
+
+    ODQuery* q = [ODQuery queryWithNode:root
+                         forRecordTypes:kODRecordTypeUsers
+                              attribute:kODAttributeTypeUniqueID
+                              matchType:kODMatchEqualTo
+                            queryValues:[NSString stringWithFormat:@"%d", uid]
+                       returnAttributes:@"dsAttrTypeNative:accountPolicyData"
+                         maximumResults:0
+                                  error:&err];
+    if (err != nullptr) {
+      TLOG << "Error with OpenDirectory query: "
+           << std::string([[err localizedDescription] UTF8String]);
+    }
+
+    // Obtain the results synchronously, not good for very large sets.
+    NSArray* od_results = [q resultsAllowingPartial:NO error:&err];
+    if (err != nullptr) {
+      TLOG << "Error with OpenDirectory results: "
+           << std::string([[err localizedDescription] UTF8String]);
+    }
+    for (ODRecord* re in od_results) {
+      if (err != nullptr) {
+        TLOG << "Error with OpenDirectory attribute: "
+             << std::string([[err localizedDescription] UTF8String]);
+      } else {
+        NSData* userPlistData =
+            [re valuesForAttribute:@"dsAttrTypeNative:accountPolicyData"
+                             error:nil][0];
```

Comment:
Do we have some assurance that this array is going to have at least one element? I think we should be checking the length before accessing index `0`.